WeScan/ImageScannerController.swift

```diff
@@ -129,25 +129,67 @@ public final class ImageScannerController: UINavigationController {
             self.blackFlashView.isHidden = true
         }
     }
+}
+
+/// Data structure containing information about a scan, including both the image and PDF version.
+public struct ImageScannerScan {
+    public var image: UIImage
+    public var pdf: Data?
     
+    mutating func rotate(by rotationAngle: Measurement<UnitAngle>) {
+        guard rotationAngle.value != 0, rotationAngle.value != 360 else { return }
+        
+        image = image.rotated(by: rotationAngle) ?? image
+        pdf = image.pdfData()
```

Comment:
Am I right or does this means that we always create an image _and_ a PDF from a scan? This would mean that if we don't use a PDF in the end, it's unneeded work done. Let me know your thoughts, but otherwise we might want to create the PDF at the moment of requesting

Reply:
Yes, right now we're always creating a PDF. It's not the best solution, so I'd love to change it, but I didn't have any better ideas? There's really no way to create it 'at the moment' because of how we're passing the `ImageScannerResults` down to the delegate. Is there a better way?

Reply:
Could we have a function on `ImageScannerScan` to generate the pdf?

Reply:
@Boris-Em So we would only give the image back with a pdf() function the host app can call? I mean it would work but it would cause a noticeable speed delay after dismissing WeScan, are we sure its the best solution? cc @AvdLee  🤔

Reply:
Not sure if this passed through, but this was my response:

> Well, at the moment a user requests a PDF, we have the image as a default. You can create a PDF at that point easily, right? That would solve a lot of unneeded PDF creations inbetween. Let me know your thoughts!

If it's causing a delay, we could show an exporting popup? It's not strange that it could take up some time to "generate PDF", right?

Reply:
This is now implemented! PDFs are now generated asynchronously on request by the host app, so there is no longer a speed difference compared to the non-PR/old version of WeScan.

Everything else remains the same, so host apps do not have to use custom PDF generation code and we can maintain and test this ourselves.